### PR TITLE
[#47] - Add user page

### LINKED ISSUE #47: Add a user page
**Is your feature request related to a problem? Please describe.**

This is a front end component, please refer to the Figma on the wiki for the current design. If there is no design, feel free to add your own to fit the criteria.

We need a user page to show a user’s details and post history. The page should be at the route “/user/:username”

**Describe the solution you'd like**
- Details section and a history section
- A MainHeader and Footer

User details section:
- Profile picture image
- Profile banner image
- Following/follower count (should also be a link to the user’s follower details page)
- Username, nickname, bio
- User joined date formatted to “Month YYYY”
- A follow button if they are a different user, or a settings button if the current logged in user is browsing their own profile. 

User history section:
- A list of post components (each in condensed view with no replies shown)

Here is some mock data for a user and a feed

```js
// mock-data.js
export const sampleUser = {
    id: 1,
    username: "username",
    nickname: "nickname",
    bio: "a biography", 
    profileBanner: "https://i.imgur.com/PcEvuMw.png",
    profilePic: "https://i.imgur.com/qZImY9j.jpg",
    followers: 11,
    following: 22,
    joinedDate: 1646627469012
}

export const sampleFeed = [
    {
        id: 1,
        content: "Post 1",
        author: sampleUser,
        parent: null,
        children: [5],
        usersLiked: 1,
        usersShared: 2,
        timestamp: 1646627469012
    }, 
    {
        id: 2,
        content: "Post 2",
        author: sampleUser,
        parent: null,
        children: [],
        usersLiked: 50,
        usersShared: 33,
        timestamp: 1646627527764
    },
    {
        id: 3,
        content: "Post 3",
        author: sampleUser,
        parent: null,
        children: [],
        usersLiked: 10,
        usersShared: 20,
        timestamp: 1646627527764
    },
]
```

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by BruceZeng1**
# Description:
Added user page at user/:username.
Using mock data found in frontend/src/pages/user/mock-data.js.
No Follow button or settings button yet due to no authentication.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](../CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] My code follows the code style of this project
- [ ] My code has been commented
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

### CODE REVIEW

**Review by skwe902 — COMMENTED** [review]
LGTM!

**Review by BraveNewWord — APPROVED** [review]
LGTM

File: frontend/src/pages/user/UserPageView.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            className={classes.banner}
+            alt="Profile Banner"
+            src={userData.profileBanner}
+        />
+        <div className={classes.pageContent}>
+            <Button
+                variant="contained"
+                disableElevation
+                className={classes.followButton}
+            >
```
Comment (nzbasic) [inline frontend/src/pages/user/UserPageView.jsx]: You should pass an empty onClick handler from the controller so we can easily add logic later

**Review by nzbasic — CHANGES_REQUESTED** [review]

**Review by harristomy — APPROVED** [review]

**Review by Billsong19 — APPROVED** [review]
LGTM, good work!

**Review by henrym25 — APPROVED** [review]

**Review by nzbasic — APPROVED** [review]

**Pull request merged**